### PR TITLE
[release-8.0] Make unix domain server timeout configurable

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: cmake/InstallJsonrpccpp.cmake [hunk 1]
```diff
@@ -46,8 +46,8 @@ set(CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
 
 ExternalProject_Add(jsonrpc-project
     PREFIX src/depends/jsonrpc
-    URL https://github.com/cinemast/libjson-rpc-cpp/archive/v1.2.0.tar.gz
-    URL_HASH SHA256=485556bd27bd546c025d9f9a2f53e89b4460bf820fd5de847ede2539f7440091
+    URL https://github.com/KaustubhShamshery/libjson-rpc-cpp/archive/v1.2.0-time-patch.tar.gz
```
Comment (bb111189) [inline cmake/InstallJsonrpccpp.cmake:49]: change to this repo https://github.com/Zilliqa/libjson-rpc-cpp

**Review by bb111189 — CHANGES_REQUESTED** [review]

File: tests/Server/Test_ScillaIPCServer.cpp [hunk 2]
```diff
@@ -43,6 +43,7 @@ BOOST_AUTO_TEST_SUITE(scillaipc)
 BOOST_AUTO_TEST_CASE(test_connection) {
   INIT_STDOUT_LOGGER();
   UnixDomainSocketServer s(SCILLA_IPC_SOCKET_PATH);
+  s.SetWaitTime(10);
```
Comment (sandipbhoir) [inline tests/Server/Test_ScillaIPCServer.cpp:46]: May be we add wait time to constants.xml

**Review by sandipbhoir — CHANGES_REQUESTED** [review]

**Comment by vaivaswatha** [comment]
I see that you're calling `SetWaitTime()` at each place that `UnixDomainSocketServer` is instantiated. An alternate to this is to derive from `UnixDomainSocketServer`, a class that sets `SetWaitTime` in its constructor. That way you only create instances of this class and don't manually set the time everytime.

**Review by sandipbhoir — APPROVED** [review]

**Review by bb111189 — APPROVED** [review]

**Comment by KaustubhShamshery** [comment]
> I see that you're calling `SetWaitTime()` at each place that `UnixDomainSocketServer` is instantiated. An alternate to this is to derive from `UnixDomainSocketServer`, a class that sets `SetWaitTime` in its constructor. That way you only create instances of this class and don't manually set the time everytime.

For now since only scilla server needs setting of this value and it is only intastiated once, the current design does not have any major disadvantage. However, the concern is valid, can be made a lower priority issue

**Pull request merged**